src/Payments.sol

```diff
@@ -838,13 +838,81 @@ contract Payments is
             string memory note
         )
     {
-        return settleRailInternal(railId, untilEpoch, false);
+        return settleRailInternal(railId, untilEpoch, false, false);
+    }
+
+    /// @notice Settles and forecloses a rail. Can only be called by the operator or payee.
+    /// This method settles the rail, credits available funds to payee (up to account.lockupSettledAt + rail.lockupPeriod),
+    /// then finalizes the rail and marks it as inactive.
+    /// @param railId The ID of the rail to settle and foreclose.
+    /// @return totalSettledAmount The total amount settled and transferred.
+    /// @return finalSettledEpoch The epoch up to which settlement was actually completed.
+    /// @return note Additional information about the settlement.
+    function settleAndForecloseRail(
+        uint256 railId
+    )
+        external
```

Comment:
this should work only upto and including the current block and only if account lockup is NOT fully settled.